### PR TITLE
[FEAT] Add XLinear

### PR DESCRIPTION
**Pull request by marcopeix**
Adds the XLinear model to neuralforecast.

**Results reproduction**
_On ETTm1, h=96, multivariate_
| MSE (paper) | MSE (ours) | MAE (paper) | MAE (ours)|
|:-------------|:------------|:-------------|:-----------|
| 0.311 | 0.327 | 0.351 | 0.351 |

_On ETTm1, h=96, univariate with features_
| MSE (paper) | MSE (ours) | MAE (paper) | MAE (ours)|
|:-------------|:------------|:-------------|:-----------|
| 0.028 | 0.034 | 0.125 | 0.138 |

### CODE REVIEW

**Review by elephaint — COMMENTED** [review]
## PR Review: XLinear Model Implementation

This PR adds the XLinear model for multivariate time series forecasting with gating mechanisms, along with AutoXLinear for hyperparameter tuning.

### Strengths
- Clean implementation following existing codebase patterns
- Good defensive validation with helpful error messages for `n_series` mismatch
- Proper warning when `channel_ff` is not a multiple of `n_series`
- AutoXLinear properly enforces `n_series` consistency across backends
- CI/CD improvements with uv caching are good for build performance

### Issues Found

**1. Missing architecture image (docs/models.xlinear.html.md:18)**
References `imgs_models/xlinear.png` but the image doesn't exist. This will show a broken link.

**2. Missing parameter documentation (neuralforecast/models/xlinear.py:80)**
`exclude_insample_y` parameter is in `__init__` but not documented in the docstring.

**3. Performance: Use broadcasting instead of `.repeat()` (neuralforecast/models/xlinear.py:319-324)**
```python
# Current (allocates new tensor each forward pass):
dec_out = dec_out * stdev[:, 0, :].unsqueeze(1).repeat(...)

# Better (uses broadcasting, no allocation):
dec_out = dec_out * stdev[:, 0:1, :]
dec_out = dec_out + means[:, 0:1, :]
```

**4. Inconsistent `.detach()` (neuralforecast/models/xlinear.py:246)**
`means` is detached but `stdev` (lines 248-250) is not. If RevIN is intended, both should be detached.

**5. Missing tests for model-specific validation (tests/test_models/test_xlinear.py)**
- The `n_series` mismatch `ValueError` (xlinear.py:237-242) is not tested
- The `channel_ff` warning (xlinear.py:174-180) is not tested

**6. AutoXLinear missing searchable hyperparameters (neuralforecast/auto.py:2564-2576)**
`default_config` doesn't include `temporal_ff`, `channel_ff`, or dropout parameters. Given the warning about `channel_ff % n_series`, this could be valuable for tuning.

**7. TimeXer breaking change (neuralforecast/models/timexer.py:186)**
`EXOGENOUS_FUTR` changed from `True` to `False` and future exogenous handling removed from `forward()`. Verify this is intentional.

**Review by elephaint — CHANGES_REQUESTED** [review]
See comments above, only the image is necessary imho (the rest is nice-to-have).
